### ISSUE TITLE
image_optim.yml

### ISSUE THREAD

**Issue by ahmgeek** [toy/image_optim#108]
I've added the config file in <code>config/</code> inside my rails app ass ```.image_optim.yml```
and also tried ```image_optim.yml``` without any hope to work !
I don't have the pleasure to put it in ```.config/``` in the deployment as its heroku.
is there anything wrong I do ?

**Comment by toy** [comment]
You need to either put the `.image_optim.yml` in the application root directory or put it in `config/image_optim.yml` and add `config.assets.image_optim.config_path = 'config/image_optim.yml'` to rails config.

**Comment by ahmgeek** [comment]
I've did that, thanks a bunch.
however, rails raised some errors due  to these configs

```
jpeg:
  jhead:
  jpegoptim:
    strip: all
    max_quality: 75
  jpegtran:
    copy_chunks: false
    progressive: true
    jpegrescan: false
  jpegrecompress:
    quality: 2
png:
  pngcrush:
    chunks: alla
    fix: false
    brute: false
    blacken: true
  optipng:
    level: 4
    interlace: false
    strip: true
  pngquant:
    quality: 100..100
    speed: 3
  advpng:
    level: 4
gif:
  gifsicle:
    interlace: false
    level: 3
    careful: false
  gifsicle:
    interlace: true
    level: 3
    careful: false
```

any configuration file around I can cheat from ?

**Comment by ahmgeek** [comment]
this is the typical error: 
```
/home/ahmgeek/.rvm/gems/ruby-2.2.2/gems/image_optim-0.21.0/lib/image_optim/config.rb:79:in `assert_no_unused_options!': unknown options {:jpeg=>{:jhead=>nil, :jpegoptim=>{:strip=>"all", :max_quality=>75}, :jpegtran=>{:copy_chunks=>false, :progressive=>true, :jpegrescan=>false}, :jpegrecompress=>{:quality=>2}}, :png=>{:pngcrush=>{:chunks=>"alla", :fix=>false, :brute=>false, :blacken=>true}, :optipng=>{:level=>4, :interlace=>false, :strip=>true}, :pngquant=>{:quality=>"100..100", :speed=>3}, :advpng=>{:level=>4}}, :gif=>{:gifsicle=>{:interlace=>true, :level=>3, :careful=>false}}} (ImageOptim::ConfigurationError)
```

**Comment by toy** [comment]
Put worker names on the root level, there is no separation by image format.

**Comment by ahmgeek** [comment]
thanks.

**Comment by dt1973** [comment]
@toy putting `config.assets.image_optim.config_path = "config/image_optim.yml"` in `config/application.rb` results in `undefined method `config_path=' for nil:NilClass (NoMethodError)`.

If this is at all possible, could we maybe document it in the README?

**Comment by zeroedin** [comment]
+1 for @dt1973 comment

**Comment by toy** [comment]
@dt1973 @zeroedin You are right, I've created #111. Also the option should be `config_paths` and not `config_path`, I've fixed it in my reply for less confusion

**Comment by toy** [comment]
@dt1973 @zeroedin Fixed in 4daabc3ad44877bb61e3cf2bb2e0b40633a2bd30 and 4f3137e9248bcce3f33561b55593fdd1dcf9a04d

**Comment by ahmgeek** [comment]
@toy thanks man *MAD HUGS*
